include/lsst/meas/astrom/ScaledPolynomialTransformFitter.h

```diff
@@ -0,0 +1,275 @@
+     *                              quadrature with source measurement errors in
+     *                              setting the uncertainty on each match.
+     *
+     *  This initializes the data catalog with the following fields:
+     *   - ref_id:             reference object ID
+     *   - src_id:             source ID from the match
+     *   - src_{x,y}:          source position in pixels
+     *   - src_{xSigma,ySigma,x_y_Cov}:  uncertainty on source positions,
+     *                         including measurement errors and the inferred
+     *                         intrinsic scatter.
+     *   - intermediate_{x,y}: reference positions in intermediate world
+     *                         coordinates.
+     *   - initial_{x,y}:      reference positions transformed by initialWcs.
+     *   - model_{x,y}:        reference positions transformed by current
+     *                         best-fit distortion.
+     *   - valid:              Flag that is set for objects that have not been
+     *                         rejected as outliers.
+     */
+    static ScaledPolynomialTransformFitter fromMatches(
+        int maxOrder,
+        afw::table::ReferenceMatchVector const & matches,
+        afw::image::Wcs const & initialWcs,
+        double intrinsicScatter
+    );
+
+    /**
+     *  Initialize a fit that inverts an existing transform by evaluating and
+     *  fitting to points on a grid.
+     *
+     *  @param[in] maxOrder         Maximum polynomial order for the fit.
+     *
+     *  @param[in] bbox             Bounding box for the grid in the input
+     *                              coordinates of the toInvert transform
+     *                              (or equivalently the output coordinates
+     *                              of the transform to be fit).
+     *
+     *  @param[in] nGridX           Number of grid points in the X direction.
+     *
+     *  @param[in] nGridY           Number of grid points in the Y direction.
+     *
+     *  @param[in] toInvert         Transform to invert
+     *
+     *  This initializes the data catalog with the following fields:
+     *   - output:  grid positions passed as input to the toInvert transform,
+     *              treated as output data points when fitting its inverse.
+     *   - input:   grid positions generated as the output of the toInvert
+     *              transform, treated as input data points when fitting its
+     *              inverse.
+     *   - model:   best-fit-transformed input points.
+     *
+     *  The updateIntrinsicScatter and rejectOutliers methods cannot be used
+     *  on fitters initialized using this method.
+     */
+    static ScaledPolynomialTransformFitter fromGrid(
+        int maxOrder,
+        afw::geom::Box2D const & bbox,
+        int nGridX, int nGridY,
+        ScaledPolynomialTransform const & toInvert
+    );
+
+    /**
+     *  Perform a linear least-squares fit of the polynomial coefficients.
+     *
```

Comment:
Done.  I've added a bit to each of these functions and a code block to the class docs describing the typical iteration pattern.